### PR TITLE
drivers: mipi_dbi: nxp_lcdic: add support for 8080 mode

### PR DESCRIPTION
**Pull request by danieldegrasse**
Enable support for 8080 mode on the NXP LCDIC peripheral. This support has been validated on the `rd_rw612_bga` board using `lcd_par_s035_8080` shield, with the following samples:
- `samples/drivers/display`
- `samples/modules/lvgl/demos`

Additionally, this PR was tested on the `frdm_mcxn947//cpu0` using the `lcd_par_s035_8080` shield with `samples/modules/lvgl/demos`.

~Depends on #77578, #77579, and #77584 for various driver fixes (commits from those PRs are currently included in this one)~
~Blocked by https://github.com/zephyrproject-rtos/zephyr/pull/78415 due a CI fix needed~

### CODE REVIEW

File: drivers/display/display_st7796s.c [hunk 1]
```diff
@@ -118,7 +118,13 @@ static int st7796s_get_pixelfmt(const struct device *dev)
...
 	 */
 	if (config->dbi_config.mode == MIPI_DBI_MODE_8080_BUS_8_BIT) {
-		if (config->madctl & ST7796S_MADCTL_BGR) {
+		/*
+		 * Similar to the handling for other interface modes,
+		 * invert the reported pixel format if rgb_is_inverted"
+		 * is enabled
+		 */
+		if (!(config->madctl & ST7796S_MADCTL_BGR) !=
+		    !config->rgb_is_inverted) {
```
Comment (decsny) [inline drivers/display/display_st7796s.c:127]:
```suggestion
                if ((bool)(config->madctl & ST7796S_MADCTL_BGR) != config->rgb_is_inverted) {
```

expression seems redundant with the double negative, also I'm not expert on the C standard but I think the first operand might need to be a boolean, at least better safe than sorry

File: boards/shields/lcd_par_s035/lcd_par_s035_8080.overlay [hunk 2]
```diff
@@ -24,8 +24,8 @@
 	gt911_lcd_par_s035: gt911-lcd_par_s035@5d {
 		compatible = "goodix,gt911";
 		reg = <0x5d>;
-		alt-addr = <0x14>;
```
Comment (decsny) [inline boards/shields/lcd_par_s035/lcd_par_s035_8080.overlay:27]: why is this deleted? might want to mention in commit message

File: drivers/display/display_st7796s.c (hunk 1 above)
Comment (danieldegrasse) [reply drivers/display/display_st7796s.c:127]: Casting certainly won't hurt. I *think* the reason the logic was written like this initially was that the `!` is going to effectively cast the value to a boolean (at least per my understanding). I'll update to make this change, and fix the logic on the lines below to match it

File: boards/shields/lcd_par_s035/lcd_par_s035_8080.overlay (hunk 2 above)
Comment (danieldegrasse) [reply boards/shields/lcd_par_s035/lcd_par_s035_8080.overlay:27]: I've added a blurb in the commit message. For anyone interested, the best I can tell this was working because the FRDM-MCXN947 board (previously the only in tree board using this display) was configuring the INT pin for the display in the MIPI-DBI pincontrol node. The pin control setting was such that when the RESET pin for the display (and touch screen) controller is toggled low, the touch screen controller will select the 0x14 alternate I2C address. This workaround wasn't working on the RW612 board, so I opted to properly set initialization order, thereby making sure the GT911 driver is able to pull the reset pin low and set the INT pin properly as needed to get the GT911 IC to select the standard 0x5D I2C address.

File: drivers/display/display_st7796s.c (hunk 1 above)
Comment (decsny) [reply drivers/display/display_st7796s.c]: i see, makes sense, still think it's not as clear of a way to write it though which is more important than saving a couple characters

**Review by decsny — DISMISSED** [review]

File: drivers/display/display_st7796s.c (hunk 1 above)
Comment (danieldegrasse) [reply drivers/display/display_st7796s.c]: Sure- I changed the logic to match your suggestion.

File: dts/bindings/mipi-dbi/nxp,lcdic.yaml [hunk 3]
```diff
@@ -23,3 +23,26 @@ properties:
     description: |
       Swap bytes while transferring on LCDIC. When set, the LCDIC will send
       the most significant byte first when using multibyte pixel formats.
+
```
Comment (jfischer-no) [inline dts/bindings/mipi-dbi/nxp,lcdic.yaml:26]: Do you want to remove "Only SPI mode is currently supported."(L:6)?

Comment (danieldegrasse) [reply dts/bindings/mipi-dbi/nxp,lcdic.yaml:26]: Yes, we should. I'll update this PR, thanks for catching that

**Review by jfischer-no — DISMISSED** [review]

**Review by decsny — DISMISSED** [review]

File: boards/nxp/rd_rw612_bga/doc/index.rst [hunk 4]
```diff
@@ -75,6 +75,59 @@ The default configuration can be found in the defconfig file:
...
+================
+
+The :ref:`adafruit_2_8_tft_touch_v2` connects to the board's Arduino headers
+directly, but some modifications are required (see
+:zephyr_file:`boards/shields/adafruit_2_8_tft_touch_v2/boards/rd_rw612_bga.overlay`
+for a list). The display sample can be built for this module like so:
+
+.. zephyr-app-commands::
+   :board: rd_rw612_bga
+   :gen-args: -DSHIELD=adafruit_2_8_tft_touch_v2
```
Comment (kartben) [inline boards/nxp/rd_rw612_bga/doc/index.rst:110]:
```suggestion
   :shield: adafruit_2_8_tft_touch_v2
```

File: boards/nxp/rd_rw612_bga/doc/index.rst [hunk 5]
```diff
@@ -75,6 +75,59 @@ The default configuration can be found in the defconfig file:
...
+
+The :ref:`lcd_par_s035` does not connect directly to the board, and must be
+connected via jumper wires. Connections and required board changes are
+described in
+:zephyr_file:`boards/shields/lcd_par_s035/boards/rd_rw612_bga.overlay`. The
+display sample can be built for the module like so:
+
+.. zephyr-app-commands::
+   :board: rd_rw612_bga
+   :gen-args: -DSHIELD=lcd_par_s035_8080
```
Comment (kartben) [inline boards/nxp/rd_rw612_bga/doc/index.rst:126]:
```suggestion
   :shield: lcd_par_s035_8080
```

File: drivers/display/display_st7796s.c [hunk 6]
```diff
@@ -118,7 +118,13 @@ static int st7796s_get_pixelfmt(const struct device *dev)
 	 * and vice versa.
 	 */
 	if (config->dbi_config.mode == MIPI_DBI_MODE_8080_BUS_8_BIT) {
-		if (config->madctl & ST7796S_MADCTL_BGR) {
+		/*
+		 * Similar to the handling for other interface modes,
+		 * invert the reported pixel format if rgb_is_inverted"
```
Comment (kartben) [inline drivers/display/display_st7796s.c:123]:
```suggestion
		 * invert the reported pixel format if "rgb_is_inverted"
```

**Review by kartben — CHANGES_REQUESTED** [review]
Minor doc comment -- we now have a "--shield" west arg

**Comment by danieldegrasse** [comment]
Rebased to pick up CI fixes

**Comment by danieldegrasse** [comment]
@kartben ping- could you take another look at this PR?

**Review by kartben — APPROVED** [review]

**Review by EmilioCBen — APPROVED** [review]

**Review by dleach02 — APPROVED** [review]
